src/api/client.ts

```diff
@@ -142,18 +63,22 @@ export class WsClient extends Emittery {
   static connect(url: URL, options?: WsClientOptions) {
     return new Promise<WsClient>((resolve, reject) => {
       const socket = new IsoWebSocket(url, options);
-      socket.onerror = (errorEvent) => {
+      socket.addEventListener("error", (errorEvent) => {
```

Comment:
`addEventListener` used throughout now, with one-off listeners for all calls returning a promise.